src/MediaWiki.ts

```diff
@@ -203,7 +201,7 @@ class MediaWiki {
     const body = await downloader.query(query);
     const entries = body.query.general;
 
-    const mainPage = entries.mainpage.replace(/ /g, self.spaceDelimiter);
+    const mainPage = entries.mainpage.replace(/ /g, '_');
```

Comment:
I don’t understand why we still have a replace() here.

Reply:
This is what we call, in the very beginning to getSiteInfo, where in response we get a mainpage. Now, eg, if we are doing the scrape for the bm.wikipedia,

The query that we request looks like:
`action=query&meta=siteinfo&format=json&siprop=general|namespaces|statistics|variables|category|wikidesc
`

Response (only relevant part of response JSON):
```
{
  mainpage: 'Nyɛ fɔlɔ',
  base: 'https://bm.wikipedia.org/wiki/Ny%C9%9B_f%C9%94l%C9%94',
  sitename: 'Wikipedia',
  logo: '//bm.wikipedia.org/static/images/project-logos/bmwiki.png',
  generator: 'MediaWiki 1.35.0-wmf.39'
}
```

So, the mainpage is with space, so we need to replace that with '_'.

This is the sample article on wikipedia:
https://bm.wikipedia.org/wiki/Latino_Amerika

This is the sample API request to fetch the article:
https://bm.wikipedia.org/api/rest_v1/page/mobile-sections/Latino_Amerika

Here is the API documentation where it is mentioned that '_' should be used instead of spaces.
https://en.wikipedia.org/api/rest_v1/#/Mobile/getSections

Reply:
The Mediawiki documentation says at https://www.mediawiki.org/wiki/API:Siteinfo:
> mainpage: The title of the main page, as found in MediaWiki:Mainpage. 1.8+

So this is the title, not the pageid, and in fact it can be customized and it is, see for WPEN:
https://en.wikipedia.org/wiki/MediaWiki:Mainpage

So, here the problem is that you are not looking at the right place IMO. Considering that there is no proper entry in the json. I would recommend to extract it from the `base` URL.

Reply:
But, the base will be encoded and look something like this:

`https://bm.wikipedia.org/wiki/Ny%C9%9B_f%C9%94l%C9%94`

So doing a operation to:
1. Extract the last segment from URl and then
2. Decoding it
is much overhead than a simple replace. Please suggest?

Reply:
you need of course to extract the id and url-decode it.